chia/wallet/wallet_state_manager.py

```diff
@@ -692,6 +712,127 @@ async def determine_coin_type(
 
         return None
 
+    def deserialize_coin_metadata(self, metadata: Optional[VersionedBlob], coin_type: CoinType) -> Any:
+        if metadata is None:
+            return None
+
+        if coin_type == CoinType.CLAWBACK:
+            return ClawbackMetadata.from_bytes(metadata.blob).to_json_dict()
+        else:
+            return metadata.blob
+
+    async def auto_claim_coins(self) -> None:
+        # Get unspent clawback coin
+        unspent_coins = await self.coin_store.get_all_unspent_coins(coin_type=CoinType.CLAWBACK)
+        current_timestamp = self.blockchain.get_latest_timestamp()
+        clawback_coins: Dict[Coin, ClawbackMetadata] = {}
+        tx_fee = uint64(self.config.get("auto_claim", {}).get("tx_fee", 0))
+        min_amount = uint64(self.config.get("auto_claim", {}).get("min_amount", 0))
+        for coin in unspent_coins:
+            if coin.metadata is None or coin.metadata.blob is None or len(coin.metadata.blob) <= 1:
+                self.log.error(f"Cannot auto claim clawback coin {coin.coin.name().hex()} since missing metadata.")
+                continue
+            metadata = ClawbackMetadata.from_bytes(coin.metadata.blob)
+            if coin.coin.amount >= min_amount and await metadata.is_recipient(self.puzzle_store):
```

Comment:
If you use `get_coin_records` instead of `get_all_unspent_coins` above you can also use `amount_filter=AmountFilter(start=uint64(min_amount))` already in the query and avoid this condition here.

Reply:
AmountFilter is not a range filter, or maybe I misunderstood it.